diff --git a/helix-term/src/ui/editor.rs b/helix-term/src/ui/editor.rs
index 6be56574..8b294a2f 100644
--- a/helix-term/src/ui/editor.rs
+++ b/helix-term/src/ui/editor.rs
@@ -1438,10 +1438,11 @@ fn handle_event(
                                     {
                                         consumed = true;
                                         Some(callback)
-                                    } else if let EventResult::Consumed(callback) =
-                                        completion.handle_event(&Event::Key(key!(Enter)), &mut cx)
-                                    {
-                                        Some(callback)
+                                    // maybe if the config is not preview...
+                                    // } else if let EventResult::Consumed(callback) =
+                                    //     completion.handle_event(&Event::Key(key!(Enter)), &mut cx)
+                                    // {
+                                    //     Some(callback)
                                     } else {
                                         None
                                     }
diff --git a/helix-term/src/ui/popup.rs b/helix-term/src/ui/popup.rs
index db77492d..82e909f3 100644
--- a/helix-term/src/ui/popup.rs
+++ b/helix-term/src/ui/popup.rs
@@ -261,7 +261,7 @@ fn handle_event(&mut self, event: &Event, cx: &mut Context) -> EventResult {
             _ => return EventResult::Ignored(None),
         };
 
-        if key!(Esc) == key && self.ignore_escape_key {
+        if (key!(Esc) == key || ctrl!('c') == key) && self.ignore_escape_key {
             return EventResult::Ignored(None);
         }
 
